tests/lib/nested.sh

```diff
@@ -9,29 +9,54 @@ SSH_PORT=8022
 MON_PORT=8888
 
 wait_for_ssh(){
-    retry=150
+    retry="${1:-120}"
+    wait="${2:-1}"
     while ! execute_remote true; do
         retry=$(( retry - 1 ))
         if [ $retry -le 0 ]; then
             echo "Timed out waiting for ssh. Aborting!"
             return 1
         fi
-        sleep 1
+        sleep $wait
     done
 }
 
 wait_for_no_ssh(){
-    retry=150
+    retry="${1:-120}"
+    wait="${2:-1}"
     while execute_remote true; do
         retry=$(( retry - 1 ))
         if [ $retry -le 0 ]; then
             echo "Timed out waiting for no ssh. Aborting!"
             return 1
         fi
-        sleep 1
+        sleep $wait
     done
 }
 
+count_ready_in_log(){
+    journalctl -u "$NESTED_VM" | grep -c "Press enter to configure"
+}
+
+wait_for_ready(){
+    pre_ready="${1:-0}"
+    retry="${2:-120}"
+    wait="${3:-2}"
+
+    while [ $(count_ready_in_log) == $pre_ready ] ; do
```

Comment:
This test works when comparing to 0, but I think this should be -le if we want to wait for more occurrences of the string (or -lt if we want at least that number of occurrences, which would make the parameter more intuitive). Maybe generalize it later to wait for any given expression?

Reply:
This code has been removed.